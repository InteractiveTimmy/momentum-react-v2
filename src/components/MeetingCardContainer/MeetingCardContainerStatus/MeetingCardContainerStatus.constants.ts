import type { Color } from './MeetingCardContainerStatus.types';

const CLASS_PREFIX = 'md-meeting-card-container-status';

const COLORS: Record<string, Color> = {
  INACTIVE: 'inactive',
  SUCCESS: 'success',
};

const DEFAULTS = {};

const STYLE = {
  wrapper: `${CLASS_PREFIX}-wrapper`,
};

export { CLASS_PREFIX, COLORS, DEFAULTS, STYLE };
