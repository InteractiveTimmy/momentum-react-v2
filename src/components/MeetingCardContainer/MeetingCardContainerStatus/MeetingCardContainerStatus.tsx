import React, { FC } from 'react';
import classnames from 'classnames';

import { STYLE } from './MeetingCardContainerStatus.constants';
import { Props } from './MeetingCardContainerStatus.types';
import './MeetingCardContainerStatus.style.scss';

/**
 * The MeetingCardContainer component.
 */
const MeetingCardContainer: FC<Props> = (props: Props) => {
  const { className, color, id, striped, style, ...otherProps } = props;

  return (
    <div
      className={classnames(className, STYLE.wrapper)}
      data-color={color}
      data-striped={striped}
      id={id}
      style={style}
      {...otherProps}
    />
  );
};

export default MeetingCardContainer;
