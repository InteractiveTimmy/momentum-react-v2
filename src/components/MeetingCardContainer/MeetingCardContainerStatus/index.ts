import { default as MeetingCardContainerStatus } from './MeetingCardContainerStatus';
import * as CONSTANTS from './MeetingCardContainerStatus.constants';
import type { Props, Color } from './MeetingCardContainerStatus.types';

export { CONSTANTS as MEETING_CARD_CONTAINER_STATUS_CONSTANTS };

export type MeetingCardContainerStatusProps = Props;
export type MeetingCardContainerStatusColor = Color;

export default MeetingCardContainerStatus;
