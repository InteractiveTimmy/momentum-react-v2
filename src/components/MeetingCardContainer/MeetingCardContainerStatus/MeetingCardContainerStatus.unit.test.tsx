import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import MeetingCardContainerStatus, {
  MEETING_CARD_CONTAINER_STATUS_CONSTANTS as CONSTANTS,
} from './';

describe('<MeetingCardContainerStatus />', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      expect.assertions(1);

      const { container } = render(<MeetingCardContainerStatus />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with class name', () => {
      expect.assertions(1);

      const className = 'example-class';

      const { container } = render(<MeetingCardContainerStatus className={className} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with id', () => {
      expect.assertions(1);

      const id = 'example-id';

      const { container } = render(<MeetingCardContainerStatus id={id} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with style', () => {
      expect.assertions(1);

      const style = { color: 'pink' };

      const { container } = render(<MeetingCardContainerStatus style={style} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with color', () => {
      expect.assertions(1);

      const color = Object.values(CONSTANTS.COLORS).pop();

      const { container } = render(<MeetingCardContainerStatus color={color} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with striped', () => {
      expect.assertions(1);

      const striped = true;

      const { container } = render(<MeetingCardContainerStatus striped={striped} />);

      expect(container).toMatchSnapshot();
    });
  });

  describe('attributes', () => {
    const testid = 'test-id';

    it('should have class attribute when className prop is provided', async () => {
      expect.assertions(1);

      const className = 'example-class';

      render(<MeetingCardContainerStatus className={className} data-testid={testid} />);

      const component = await screen.findByTestId(testid);

      expect(component.classList.contains(className)).toBe(true);
    });

    it('should have id attribute when id prop is provided', async () => {
      expect.assertions(1);

      const id = 'example-id';

      render(<MeetingCardContainerStatus id={id} data-testid={testid} />);

      const component = await screen.findByTestId(testid);

      expect(component.id).toBe(id);
    });

    it('should have style attribute when style prop is provided', async () => {
      expect.assertions(1);

      const style = { color: 'pink' };

      render(<MeetingCardContainerStatus style={style} data-testid={testid} />);

      const component = await screen.findByTestId(testid);

      expect(component.style.color).toBe(style.color);
    });

    it('should have data-color attribute when color prop is provided', async () => {
      expect.assertions(1);

      const attribute = 'data-color';
      const color = Object.values(CONSTANTS.COLORS).pop();

      render(<MeetingCardContainerStatus color={color} data-testid={testid} />);

      const component = await screen.findByTestId(testid);

      expect(component.getAttribute(attribute)).toBe(color);
    });

    it('should have data-striped attribute when striped prop is provided', async () => {
      expect.assertions(1);

      const attribute = 'data-striped';
      const striped = true;

      render(<MeetingCardContainerStatus striped={striped} data-testid={testid} />);

      const component = await screen.findByTestId(testid);

      expect(component.getAttribute(attribute)).toBe(`${striped}`);
    });
  });
});
