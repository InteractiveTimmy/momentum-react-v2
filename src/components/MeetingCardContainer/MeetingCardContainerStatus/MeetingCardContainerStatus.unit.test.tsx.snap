// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MeetingCardContainerStatus /> snapshot should match snapshot 1`] = `
<div>
  <div
    class="md-meeting-card-container-status-wrapper"
  />
</div>
`;

exports[`<MeetingCardContainerStatus /> snapshot should match snapshot with class name 1`] = `
<div>
  <div
    class="example-class md-meeting-card-container-status-wrapper"
  />
</div>
`;

exports[`<MeetingCardContainerStatus /> snapshot should match snapshot with color 1`] = `
<div>
  <div
    class="md-meeting-card-container-status-wrapper"
    data-color="success"
  />
</div>
`;

exports[`<MeetingCardContainerStatus /> snapshot should match snapshot with id 1`] = `
<div>
  <div
    class="md-meeting-card-container-status-wrapper"
    id="example-id"
  />
</div>
`;

exports[`<MeetingCardContainerStatus /> snapshot should match snapshot with striped 1`] = `
<div>
  <div
    class="md-meeting-card-container-status-wrapper"
    data-striped="true"
  />
</div>
`;

exports[`<MeetingCardContainerStatus /> snapshot should match snapshot with style 1`] = `
<div>
  <div
    class="md-meeting-card-container-status-wrapper"
    style="color: pink;"
  />
</div>
`;
